docs/tutorial/onchain-rln-relay-chat2.md

```diff
@@ -0,0 +1,242 @@
+Note that the required `0.001` ethers correspond to the registration fee, 
+however, you still need to have more funds in your account to cover the cost of the transaction gas fee.
+
+
+
+## Overview
+Figure 1 provides an overview of the interaction of the chat2 clients with the test fleets and the membership contract. 
+At a high level,  when a chat2 client is run with Waku-RLN-Relay mounted in on-chain mode, it creates RLN credential (i.e., an identity key and an identity commitment key) and 
+sends a transaction to the membership contract to register the corresponding membership identity commitment key.
+This transaction will also transfer `0.001` Ethers to the contract as membership fee.
+This amount plus the transaction fee will be deducted from the supplied Goerli account. 
+Once the transaction is mined and the registration is successful, the registered credential will get displayed on the console of your chat2 client.
+You may copy the displayed RLN credential and reuse them for the future execution of the chat2 application.
+Proper instructions in this regard is provided in the following [section](#how-to-reuse-rln-credential).
+If you choose not to reuse the same credential, then for each execution, a new registration will take place and more funds will get deducted from your Goerli account.
+Under the hood, the chat2 client constantly listens to the membership contract and keeps itself updated with the latest state of the group.
+
+In the following test setting, the chat2 clients are to be connected to the Waku test fleets as their first hop. 
+The test fleets will act as routers and are also set to run Waku-RLN-Relay over the same pubsub topic and content topic as chat2 clients i.e., the default pubsub topic of `/waku/2/default-waku/proto` and the content topic of `/toy-chat/2/luzhou/proto`. 
+Spam messages published on the said combination of topics will be caught by the test fleet nodes and will not be routed.
+Note that spam protection does not rely on the presence of the test fleets.
+In fact, all the chat2 clients are also capable of catching and dropping spam messages if they receive any.
+You can test it by connecting two chat2 clients (running Waku-RLN-Relay) directly to each others and see they can spot each others' spam activities.
+
+ ![](./imgs/rln-relay-chat2-overview.png)
+ Figure 1.
+
+# Set up
+## Build chat2
+First, build chat2 with the RLN flag set to true.
+
+```
+make chat2 RLN=true
+```
+
+## Set up a chat2 client
+
+Run the following command to set up your chat2 client. 
+
+```
+./build/chat2  --fleet:test --content-topic:/toy-chat/2/luzhou/proto --rln-relay:true --rln-relay-dynamic:true --eth-mem-contract-address:0x5DE1Fb10345Ef1647629Df30e6C397297Da09A1d  --eth-account-address:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx --eth-account-privatekey:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx  --eth-client-address:xxxx  --ports-shift=1  
```

Comment:
Thanks @staheri14 for PR. Contract address is incorrect. Needs to be `0x4252105670fE33D2947e8EaD304969849E64f2a6`

Reply:
done in 0feab5efc211dfd2570144c50e7282bba4d80237